api/admin/controller/search_service_self_tests.py

```diff
@@ -0,0 +1,45 @@
+from nose.tools import set_trace
+import flask
+from flask import Response
+from flask_babel import lazy_gettext as _
+from api.admin.problem_details import *
+from core.model import (
+    ExternalIntegration
+)
+from core.external_search import ExternalSearchIndex
+from core.tests.test_external_search import ExternalSearchTest
+
+from core.selftest import HasSelfTests
+from core.util.problem_detail import ProblemDetail
+from api.admin.controller.self_tests import SelfTestsController
+
+class SearchServiceSelfTestsController(SelfTestsController, ExternalSearchTest):
+
+    def __init__(self, manager):
+        super(SearchServiceSelfTestsController, self).__init__(manager)
+        self.missing_id_error = MISSING_SEARCH_SERVICE_IDENTIFIER
```

Comment:
I don't think the specific error is super important here, since this isn't an error a normal user is likely to see. I think you could use MISSING_SERVICE wherever you would use MISSING_SEARCH_SERVICE_IDENTIFIER or MISSING_COLLECTION_IDENTIFIER, and get rid of both of those more specific documents. If you want to be more specific about what's missing, you could return something like `MISSING_SERVICE.detailed(self.missing_id_error)` and then have `missing_id_error` be a localized string like `_("Missing search service identifier")`.

This isn't a big deal, but it would let us get rid of a couple problem detail documents.

Reply:
MISSING_SERVICE seems to me like too different a type of error (it's a 404 error for if the service you looked up couldn't be found; these are 400 errors for if you somehow didn't pass in an identifier, so the lookup couldn't be performed in the first place).  I'm making a generic MISSING_IDENTIFIER error, which we can probably also use in other places at some point, and abandoning the idea of using the detailed `self.missing_id_error` (since the user probably won't see it anyway); what do you think?

Reply:
That sounds right to me.